signer/remote_signer.go

```diff
@@ -0,0 +1,80 @@
+package signer
+
+import (
+	"crypto/x509"
+	"encoding/json"
+
+	"github.com/cloudflare/cfssl/config"
+	cferr "github.com/cloudflare/cfssl/errors"
+)
+
+// A RemoteSigner represents a CFSSL instance running as signing server.
+// fulfills the Signer interface
+type RemoteSigner struct {
+	policy *config.Signing
+}
+
+// NewRemoteSigner creates a new RemoteSigner directly from a
+// signing policy.
+func NewRemoteSigner(policy *config.Signing) (*RemoteSigner, error) {
+	if policy != nil {
+		return &RemoteSigner{policy: policy}, nil
+	} else {
+		return nil, cferr.New(cferr.PolicyError,
+			cferr.InvalidPolicy)
+	}
+}
+
+// Sign sends a signature request to the remote CFSSL server,
+// receiving a signed certificate or an error in response. The hostname,
+// csr, and profileName are used as with a local signing operation, and
+// the label is used to select a signing root in a multi-root CA.
+func (s *RemoteSigner) Sign(req SignRequest) (cert []byte, err error) {
+	jsonData, err := json.Marshal(req)
+	if err != nil {
+		return nil, cferr.Wrap(cferr.APIClientError, cferr.JSONError, err)
+	}
+
+	var profile *config.SigningProfile = nil
```

Comment:
The `= nil` is extraneous and can be removed (nil is the default value for `*SigningProfile`).